pandas/_libs/reduction.pyx

```diff
@@ -491,3 +505,83 @@ cdef class BlockSlider:
         mgr.blocks = self.blocks
         mgr._blklocs = self.orig_blklocs
         mgr._blknos = self.orig_blknos
+
+
+cdef class ArraySlider(FrameSlider):
+    """
+    Only capable of sliding on axis=0
+    """
+    cdef:
+        object frame, index
+        list arrays, orig_arrays
+        # ndarray values
+        Slider idx_slider
+        char **base_ptrs
+        int narrays
+        Py_ssize_t i
+
+    def __init__(self, object frame):
+        self.frame = frame
+        self.dummy = frame[:0]
+        self.index = self.dummy.index
+
+        # GH#35417 attributes we need to restore at each step in case
+        #  the function modified them.
+        self.orig_arrays = self.dummy._mgr.arrays
+        self.arrays = list(self.dummy._mgr.arrays)
```

Comment:
BlockSlider defines `self.blk_values = [block.values for block in self.dummy._mgr.blocks]` which i think can now use the arrays property to match self.arrays here